### PR TITLE
feat: compact numbers

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #1990 

### Description:

Adds `compact` yml configuration:
```
            total_in_thousands_of_euros:
              type: sum
              format: 'eur'
              compact: thousands
```

Supported values:
- `K` or `thousands` or `thousand` 
- `M` or `millions` or `million`
- `B` or `billions` or `billion` 
- `T` or `trillions` or `trillion`

Rounding defaults to `2` when there is no `round` or `format` value. 
Screenshot 2022-11-02 at 15 22 44

Note that numbers are rounded up:
5.6 -> 5.6
5.56 -> 5.56 
5.556 -> 5.56 ( rounded up )
5.5556 -> 5.56   ( rounded up )
5.99999 -> 6 ( rounded up )

TODO for next PRs:
- update docs
- refactor to rename `numberStyle` to `compact`
- refactor `formatValue` to have value as the first arg

### CODE REVIEW

**Comment by TuringLovesDeathMetal** [comment]
@ZeRego this is great!

two questions:
1. I think you wrote G instead of `B` above?
2. for this: "Rounding defaults to 2 when there is no format value." Do you mean when there's no `round` value?

**Comment by TuringLovesDeathMetal** [comment]
ah, ya, I think we should stick to `B` for billion. (I feel like G is for Giga, which is SI units - but people don’t use SI units in business…)

**Comment by owlas** [comment]
`/cloudy list <https://github.com/lightdash/lightdash/issues/2222>`

File: packages/frontend/src/hooks/echarts/useEcharts.ts [hunk 1]
```diff
@@ -375,6 +376,7 @@ const getSimpleSeries = ({
                             formats[yFieldHash].format,
                             formats[yFieldHash].round,
                             value?.value?.[yFieldHash],
```
Comment (IrakliJani) [inline packages/frontend/src/hooks/echarts/useEcharts.ts:378]: out of scope but can we move value as the first argument?

File: packages/common/src/utils/formatting.ts [hunk 2]
```diff
@@ -129,45 +130,36 @@ function roundNumber(
...
-                round,
-                format,
-                numberStyle,
-            )}B`;
-        default:
-            return `${new Intl.NumberFormat('en-US').format(Number(value))}`;
+            )}${numberStyleConfig.suffix}`;
+        }
     }
+    return `${new Intl.NumberFormat('en-US').format(Number(value))}`;
```
[the remark this answers is not in the document]
Comment (ZeRego) [reply packages/common/src/utils/formatting.ts:155]: Note that this code is also executed in the server so we would need that information there. 
We will eventually need this to be configurable at the org and user level and similar for date formatting.

File: packages/frontend/src/hooks/echarts/useEcharts.ts (hunk 1 above)
Comment (ZeRego) [reply packages/frontend/src/hooks/echarts/useEcharts.ts:378]: Added to the PR todo list after this one is merged.

**Review by IrakliJani — APPROVED** [review]

**Pull request merged**